include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -115,14 +115,14 @@ namespace
   /**
    * Helper class containing the cell-wise prolongation operation.
    */
-  template <int dim, typename Number>
+  template <int dim, typename Number, typename Number2>
   class CellProlongator
   {
   public:
     CellProlongator(const AlignedVector<Number> &prolongation_matrix,
                     const AlignedVector<Number> &prolongation_matrix_1d,
-                    const Number                *evaluation_data_coarse,
-                    Number                      *evaluation_data_fine)
+                    const Number2               *evaluation_data_coarse,
+                    Number2                     *evaluation_data_fine)
```

Comment:
Do you actually need both `Number` and `Number2` as template arguments? At least at the call sites you modify below, `Number` is always `double`.

Reply:
My hope is to later use it with either `float` or `double`, depending on `VectorType::value_type` and with the right choice for complex numbers. I can change it for now if you prefer.

Reply:
If you've got plans for it, let's leave it as is.